### ISSUE TITLE
Synchronous version of getParameter methods don't work with same options as async

### ISSUE THREAD

**Issue by jayalfredprufrock** [vandium-io/aws-param-store#14]
So I've been happily using the async versions of `getParameter` and `getParametersByPath` without issue like this:

```
const p = await ssm.getParameter(path, {
      credentials: new AWS.SharedIniFileCredentials({profile: environment})
});
console.log(p);
```

But when I use the sync version I get a "missing credentials in config" error.
```
const p = ssm.getParameterSync(path, {
      credentials: new AWS.SharedIniFileCredentials({profile: environment})
});
console.log(p)
```

I'm guessing it has to do with how SharedIniFileCredentials is resolved or surely this would have already been reported. Not sure how most people are passing their credentials to this library...

**Comment by richardhyatt** [comment]
It does not work on the sync call because the parameters are being serialized into a JSON object and executed in a spawned process, unlike the async where it's executed in the current process. I originally wrote the module to load SSM values for code executing in AWS Lambda. I'd be open to a PR to improve the functionality.

**Comment by sappusaketh** [comment]
I do have the same issue I am using serverless express framework, localstack for testing it locally and I configured my aws-sdk on local to use localstack credentials via node-env. I dont know why this package is not using those credentials and I am getting the following error `"Error: The security token included in the request is invalid."`

**Comment by elipeters** [comment]
You can use sync if you install awscli and set the credentials there.
`apt install awscli`
`aws configure`
Then you would have to attach the required policy/s to the IAM user. I just set the **getParametersByPath** by digging into **Systems Manager**

**Comment by richardhyatt** [comment]
Fixed in 4.0.0 - the sync implementation copies the profile name that is currently configured

**Comment by javy1103** [comment]
Why is this closed if version 4.0.0 hasnt been released?
